File: Data Structures & Algorithms/products-of-array-discluding-self/submission-4.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        ## edgecase
        if not nums:
            return []
        
        result = [1 for i in range(len(nums))]

        pre = 1
        for i in range(len(nums)):
            result[i] = pre
            pre = pre * nums[i]
        
        post = 1
        for i in range(len(nums) - 1, -1, -1):
            result[i] = result[i] * post
            post = post * nums[i]

        return result
        
        

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-4.cs
public class Solution {
    public int LongestConsecutive(int[] nums) {
        int result = 0;

        // edgecase
        if(nums == null) {
            return result;
        }

        var hs = new HashSet<int>(nums);

        foreach(var num in hs) {
            if(!hs.Contains(num -1)) {
                int current = 1;
                int find = num;

                while(hs.Contains(find + 1)) {
                    find++;
                    current++;
                }

                result = Math.Max(current, result);
            }
        }

        return result;
    }
}
